Xamarin.Forms.Platform.Android/Anticipator.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Xamarin.Forms.Platform.Android
+{
+	public class Anticipator
+	{
+		const int ThreadLifeTimeSeconds = 5;
+		readonly static TimeSpan LoopTimeOut = TimeSpan.FromSeconds(ThreadLifeTimeSeconds);
+
+		readonly Thread[] _threads;
+		readonly AutoResetEvent[] _signals;
+		readonly ConcurrentQueue<Action> _actions;
+
+		internal Anticipator()
+		{
+			_actions = new ConcurrentQueue<Action>();
+
+			_threads = new Thread[Environment.ProcessorCount];
+			_signals = new AutoResetEvent[Environment.ProcessorCount];
+
+			_signals[0] = new AutoResetEvent(true);
+			_threads[0] = new Thread(Loop);
+			_threads[0].Start(_signals[0]);
+
+			Anticipate(() =>
+			{
+				for (var i = 1; i < _threads.Length; i++)
+				{
+					_signals[i] = new AutoResetEvent(true);
+					_threads[i] = new Thread(Loop);
+					_threads[i].Start(_signals[i]);
+				}
+			});
+		}
+
+		void Loop(object argument)
+		{
+			var signal = (AutoResetEvent)argument;
+
+			while (signal.WaitOne(LoopTimeOut))
+			{
+				// process actions
+				while (_actions.Count > 0)
+				{
+					if (!_actions.TryDequeue(out Action action))
+						continue; // lost race to dequeue
+
+					action();
+				}
+			}
+		}
+
+		void Signal()
+		{
+			for (var i = 0; i < _signals.Length; i++)
+			{
+				if (_signals[i] != null)
+					_signals[i].Set();
+			}
+		}
+
+		public void Anticipate(Action action)
+		{
+			if (action == null)
+				throw new ArgumentNullException(nameof(action));
+
+			_actions.Enqueue(action);
+
+			Signal();
+		}
+
+		public void AnticipateClassConstruction(Type type)
+		{
+			Anticipate(() => RuntimeHelpers.RunClassConstructor(typeof(Resource.Layout).TypeHandle));
```

Comment:
you probably should use the `type` parameter here, instead of the hardcoded Layout reference

Reply:
Oh? I'm not familiar. I wanna run the `.cctor` of `Reosurce.Layout`. What's the alternative to that than this?

Reply:
don't you want to run the `.cctor` on the `type` being passed as method parameter ??

because later on, you invoke this with `anticipator.AnticipateClassConstruction(typeof(Resource.Layout));`

Reply:
That'll help...